scripts/setup_lib/git_ops.sh

```
@@ -1,13 +1,39 @@
-    if [ -d ".git" ] && [ -d "$(git rev-parse --show-toplevel 2>/dev/null)/hypr" ]; then
-        echo "Running from Git repository. Attempting to update..." >&2
-        local_dotfiles_source_dir="$(git rev-parse --show-toplevel)"
+    local local_temp_clone_dir="" # Will store the path if a temporary clone is made
 
+    # Attempt to get the top-level directory of the current git repository.
+    local git_root
+    git_root=$(git rev-parse --show-toplevel 2>/dev/null)
+
+    # Check if 'git rev-parse' succeeded and the '/hypr' subdirectory exists at the git root.
+    # This confirms we are in the correct dotfiles repository.
+    if [ $? -eq 0 ] && [ -n "$git_root" ] && [ -d "$git_root/hypr" ]; then
```

Shellcheck (0.10.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)